gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy

```diff
@@ -227,6 +227,28 @@ class PipelineController {
     }
   }
 
+  @ApiOperation(value = "Trigger a pipeline execution")
+  @RequestMapping(value = "/v2/{application}/{pipelineNameOrId:.+}", method = RequestMethod.POST)
+  HttpEntity invokePipelineConfigViaEcho(@PathVariable("application") String application,
+                                         @PathVariable("pipelineNameOrId") String pipelineNameOrId,
+                                         @RequestBody(required = false) Map trigger) {
+    trigger = trigger ?: [:]
+    trigger.user = trigger.user ?: AuthenticatedRequest.getSpinnakerUser().orElse('anonymous')
+    trigger.notifications = trigger.notifications ?: []
+
+    RequestContext.setApplication(application)
+    try {
+      def body = pipelineService.triggerViaEcho(application, pipelineNameOrId, trigger)
+      return new ResponseEntity(body, HttpStatus.ACCEPTED)
+    } catch (NotFoundException e) {
+      throw e
+    } catch (e) {
+      log.error("Unable to trigger pipeline (application: {}, pipelineId: {})",
+        value("application", application), value("pipelineId", pipelineNameOrId), e)
+      return new ResponseEntity([message: e.message], new HttpHeaders(), HttpStatus.UNPROCESSABLE_ENTITY)
```

Comment:
Normally we'd throw an exception for this (to ensure a consistent error response vs explicitly setting `message`).

Not sure that `UNPROCESSABLE_ENTITY` is better than `BAD_REQUEST`? If we want a bad request, we could just throw a `InvalidRequestException`.

Reply:
I'd just done what was in the prior endpoint, but good point given that I'm writing a new endpoint I may as well clean this up.